### ISSUE TITLE
Provide IP Address in Duo Request

### ISSUE THREAD

**Issue by gw-uber** [hashicorp/vault#18662]
We are looking for Vault to provide the IP address of the request coming from Vault to a Duo challenge after successful LDAP authentication. 

We are trying to implement IP address restrictions for our instance of Vault. However, when a Duo challenge request is made, there is no IP address (IPv4 or v6) passed with that request (see screenshot from Duo audit logs, Vault sends 0.0.0.0) 
![no-ip-address](https://user-images.githubusercontent.com/111926560/211731589-e46ac56e-5bfd-45a7-a6c0-58708348ab60.png). We are requesting that Vault send it's host system's IP address (ex: our AWS cluster where Vault is deployed) with any Duo request so that it can be recorded in Duo's logs and if not on the Duo whitelist, blocked.

This is a p0 blocker for us as we need Vault to send the IP address of the host/requesting system to Duo so our Duo api can record the incoming IP address from these access requests, log them, and filter those that are not in the range that we have whitelisted.

The desired outcome for this FR is that when a Duo mfa challenge happens, the corresponding request, with the Duo passcode, that comes from our Vault instance includes the IP address of the host system so that it can a) be blocked if it is not in the whitelisted range and b) be recorded in the Duo audit logs. Currently there is no workaround for this.

We have [this PR](https://github.com/hashicorp/vault/issues/15337). There is also a similar PR from Duo: [here](["Networks for API access" feature for Auth API](https://duo--c.na147.visual.force.com/servlet/servlet.Integration?lid=0660g000000btV8&ic=1&linkToken=VmpFPSxNakF5TXkwd01TMHhNMVF3TURveE5Ub3dOQzQzTXpSYSxNYzBHeDVHaUhsUnJOYXd1eGRkWDcyLFlXWmtNR0po))

tl;dr: We want Vault to include it's source IP address when sending any Duo request

**Comment by hsimon-hashicorp** [comment]
This has been addressed in: https://github.com/hashicorp/vault/pull/18811
Thanks for bringing it up!
